File: September 2024/September 17/beautifulTriplets.php
<?php

function beautifulTriplets($d, $arr){
    $answer = 0;

    for($i = 0; $i < count($arr); $i++){
        for($j = $i+1; $j < count($arr); $j++){
            if ($arr[$j]-$arr[$i]=== $d){
                for($k = $j+1; $k < count($arr); $k++){
                    if($arr[$k]-$arr[$j]===$d){
                        $answer++;
                    }
                }
            }
        }
    }

    return $answer;
}

echo beautifulTriplets(3, [1, 2, 4, 5, 7, 8, 10]);

File: September 2024/September 17/beautiful_triplets.py
def beautiful_triplets(d, lst):
    answer = 0

    for x in range(len(lst)):
        for y in range(len(lst)):
            if lst[y]-lst[x] == d:
                for z in range(len(lst)):
                    if lst[z] - lst[y]== d:
                        answer += 1
    
    return answer

print(beautiful_triplets(3, [1, 2, 4, 5, 7, 8, 10]))